xero-payroll-au.yaml

```diff
@@ -3381,6 +3383,18 @@ components:
           $ref: '#/components/schemas/OpeningBalances'
         TaxDeclaration: 
           $ref: '#/components/schemas/TaxDeclaration'
+        IncomeType: 
+          $ref: '#/components/schemas/IncomeType'
+        EmploymentType: 
+          $ref: '#/components/schemas/EmploymentType'
+        CountryOfResidence:
```

Comment:
Should this be an enum instead with a new `$ref: '#/components/schemas/CountryCode'`?
See accounting yaml file for reference

Reply:
@GeorgeBillsXero Think this one should be made an enum like in the accounting yaml? Nothing in our docs indicates the all the acceptable values for this field

Reply:
Yeah, probably. Just trying to think through the implications. It'd make life simpler for clients to know acceptable values. If we ever add a country having this as an enum would make that breaking for clients (since the new value wouldn't deserialize) and that does occasionally happen (e.g. South Sudan was in our lifetime) but rarely enough that I think we can assume it doesn't. Probably that'd be breaking anyway since the client would surely be deserializing into some enum value regardless of what we spec.

The values we support are all in `CodeTable.Country` if you're looking to add them, they're not actually the same values as accounting uses.

We support, and accounting don't:

- AX (Aland Islands)
- BV (Bouvet Island)
- GF (French Guiana)
- GP (Guadeloupe)
- GS(South Georgia and the South Sandwich Islands)
- HM (Heard Island and McDonald Islands)
- MQ (Martinique)
- NF (Norfolk Island)
- TF (French Southern Territories)
- UM (United States Minor Outlying Islands)

Some of those - e.g. Heard Island and McDonald Islands - definitely aren't countries, and yet they have their own country code. 🤷‍♂️ 

Accounting support, and we don't:

-  SS (South Sudan)
    -  We really should support South Sudan, it's been over a decade.
-  SX (Sint Maarten "Dutch part")
-  VW (???)
-  XK ("temporary country code for Kosovo" according to Wikipedia)

Reply:
Thanks! Resolved by converting `CountryOfResidence` into an enum